pkg/products/rhssocommon/reconciler.go

```diff
@@ -175,11 +176,22 @@ func (r *Reconciler) CleanupKeycloakResources(ctx context.Context, inst *integre
 // this should use the same valid certs as the cluster itself but for some reason the
 // signing operator gives out self signed certs
 // to circumvent this we create another keycloak route with edge termination
-func (r *Reconciler) CreateKeycloakRoute(ctx context.Context, serverClient k8sclient.Client, config config.ConfigReadable, ssoCommon *config.RHSSOCommon) (integreatlyv1alpha1.StatusPhase, error) {
-	// We need a route with edge termination to serve the correct cluster certificate
+func (r *Reconciler) CreateKeycloakRoute(ctx context.Context, serverClient k8sclient.Client, config config.ConfigReadable, ssoCommon *config.RHSSOCommon, installation *integreatlyv1alpha1.RHMI) (integreatlyv1alpha1.StatusPhase, error) {
+
+	// For now, edge termination is used in order to support future plans for implementing user-sso rate limiting for RHOAM.
+	var (
+		terminationType = routev1.TLSTerminationReencrypt
+		hostName        = "keycloak-edge"
+	)
+
+	if config.GetNamespace() == common.RHSSOUserProductOperatorNamespace && r.Installation.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
```

Comment:
Checking the 
`config.GetNamespace() == common.RHSSOUserProductOperatorNamespace`
seems like a bad pattern, maybe just make terminationType and hostName parameters of this function.

Reply:
Thanks @matskiv, let me know if the recent change is better.